src/storage-types/src/connections/aws.rs

```diff
@@ -231,6 +241,9 @@ impl AwsAssumeRole {
         if let Some(external_id) = external_id {
             credentials = credentials.external_id(external_id);
         }
+        if let Some(region) = region {
+            credentials = credentials.region(region);
+        }
```

Comment:
And especially spooked that we need to override the region in the AssumeRole builder. This feels like something is wrong!

Reply:
I believe the first one on line 226 is required, but I agree that its cursed (perhaps fixed by the pr you linked above?)

I followed https://github.com/MaterializeInc/materialize/blob/main/src/persist/src/s3.rs#L134-L137, and based my understanding on seeing this: https://docs.rs/aws-config/0.55.1/src/aws_config/lib.rs.html#563-567

Its possible the second `AssumeRoleProdider` doesn't need it, do you want me to try that?

Reply:
pushed a commit removing this second one to test this, after the build finishes ill test in staging

Reply:
rip, its required

Reply:
You didn't bump the AWS SDK version did you?

Reply:
@benesch I didn't, because the next bump is from 0.x to 1.x and I don't want to block this pr on that change